### PR TITLE
458 support site when changing courses display vacancy status

### PR DESCRIPTION
**Pull request by JonF1**
## Context
The support agent should be aware when they are moving a candidate to a course with no vacancies. It requires consent from all parties.

## Changes proposed in this pull request
* A warning message is presented to support agent. With a checkbox for the agent to check that they understand they are moving the candidate to a course with no vacancies
* For both changing a course choice when the candidate is 'awaiting_provider_decision' and for an offered course when they are 'pending_conditions'

## Guidance to review
Try to move a candidate to a course with no vacancies in support interface in both states

## Link to Trello card
https://trello.com/c/1UccAwjM/458-support-site-when-changing-courses-display-vacancy-status

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by JonF1** [comment]
[Support site: when changing courses, display vacancy status](https://trello.com/c/1UccAwjM/458-support-site-when-changing-courses-display-vacancy-status)

File: app/views/support_interface/application_forms/application_choices/change_offered_course/confirm_offered_course_option.html.erb [hunk 1]
```diff
@@ -39,6 +39,13 @@
         <%= f.govuk_check_box :accept_guidance, true, multiple: false, label: { text: 'I have read the guidance' }, link_errors: true %>
       <% end %>
 
+      <% if flash["warning"] == I18n.t('support_interface.errors.messages.course_full_error') %>
```
Comment (tomas-stefano) [inline app/views/support_interface/application_forms/application_choices/change_offered_course/confirm_offered_course_option.html.erb:42]: Can we use another way to verify the course full than a comparison with flash against a locale? Like you rescue the Course Full in the controller and add to a variable or check against the course op object that the user want to save?

File: app/views/support_interface/application_forms/courses/edit.html.erb [hunk 2]
```diff
@@ -47,6 +47,13 @@
         <%= f.govuk_check_box :accept_guidance, true, multiple: false, label: { text: 'I have read the guidance' }, link_errors: true %>
       <% end %>
 
+      <% if flash["warning"] == I18n.t('support_interface.errors.messages.course_full_error') %>
```
Comment (tomas-stefano) [inline app/views/support_interface/application_forms/courses/edit.html.erb:50]: same as above

File: app/views/support_interface/application_forms/application_choices/change_offered_course/confirm_offered_course_option.html.erb (hunk 1 above)
Comment (carlosmartinez) [reply app/views/support_interface/application_forms/application_choices/change_offered_course/confirm_offered_course_option.html.erb:42]: agreed, the controller could assign something along the lines of `@show_course_change_confirmation`

**Comment by carlosmartinez** [comment]
other than that one change request, looks great!

File: app/views/support_interface/application_forms/application_choices/change_offered_course/confirm_offered_course_option.html.erb (hunk 1 above)
Comment (JonF1) [reply app/views/support_interface/application_forms/application_choices/change_offered_course/confirm_offered_course_option.html.erb]: Thanks, fixed in f2b6b2

**Review by tomas-stefano — APPROVED** [review]

**Review by carlosmartinez — APPROVED** [review]

**Pull request merged**